internal/driver/webui.go

```diff
@@ -179,11 +179,20 @@ func defaultWebServer(args *plugin.HTTPServerArgs) error {
 	// https://github.com/google/pprof/pull/348
 	mux := http.NewServeMux()
 	mux.Handle("/ui/", http.StripPrefix("/ui", handler))
-	mux.Handle("/", http.RedirectHandler("/ui/", http.StatusTemporaryRedirect))
+	mux.Handle("/", redirectWithQuery("/ui"))
 	s := &http.Server{Handler: mux}
 	return s.Serve(ln)
 }
 
+func redirectWithQuery(url string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if queryStr := r.URL.Query().Encode(); queryStr != "" {
+			url = url + "?" + queryStr
```

Comment:
Thanks for making this fix, I realized I had an error in my correction (really sorry about that).

Looking at the code now, it looks a bit risky to reassign the value of url within a nested function. Probably better to go back to something more like you originally had, so like:
```
urlWithQuery := url
if ... {
....
}
http.Redirect(w, r, urlWithQuery, http.StatusTemporaryRedirect)
```

Reply:
no worries, updated it